### PR TITLE
Day Calculator

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/1031

*Description of changes:*
Added a new modal for day calculator on the engagement form.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by tiagograf-AOT — APPROVED** [review]

File: met-web/src/components/common/Modals/DayCalculator/index.tsx [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+    // calculator
+    const calculator = () => {
+        let calcStartDate = startDate;
+        let calcEndDate = endDate;
+        let calcNumberOfDays = numberOfDays;
+        if (selectedOption?.label == options[0].label) {
+            if (startDate && endDate && !numberOfDays) {
+                calcNumberOfDays = dayjs(endDate).diff(dayjs(startDate), 'days');
+            } else if (startDate && !endDate && numberOfDays) {
+                calcEndDate = dayjs(startDate).add(numberOfDays, 'day').format('YYYY-MM-DD');
```
Comment (saravanpa-aot) [inline met-web/src/components/common/Modals/DayCalculator/index.tsx:83]: Please make YYYY-MM-DD as a constant

File: met-web/src/components/common/Modals/DayCalculator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+            } else if (!startDate && endDate && numberOfDays) {
+                calcStartDate = dayjs(endDate).subtract(numberOfDays, 'day').format('YYYY-MM-DD');
+            }
+        }
+
+        if (selectedOption?.label == options[1].label) {
```
Comment (saravanpa-aot) [inline met-web/src/components/common/Modals/DayCalculator/index.tsx:89]: Please write enough comments about the logic ;
One of the better style is to use meaningful variable names 
https://www.loadingdeveloper.com/writing-cleaner-code-making-cleaner-if-statements/

**Review by saravanpa-aot — APPROVED** [review]
Please write lot of comments to explain the logic ; lot of if statements are hard to read generally , so comments will help the next dev

File: met-web/src/components/common/Modals/DayCalculator/index.tsx [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+        });
+    };
+
+    // calculator
+    const calculator = () => {
+        let calcStartDate = startDate;
+        let calcEndDate = endDate;
+        let calcNumberOfDays = numberOfDays;
+        if (selectedOption?.label == options[0].label) {
+            if (startDate && endDate && !numberOfDays) {
```
Comment (jadmsaadaot) [inline met-web/src/components/common/Modals/DayCalculator/index.tsx:80]: Can we try breaking every logic block into separate function with meaningful name like https://github.com/ryanmcdermott/clean-code-javascript#functions-should-do-one-thing

Comment (jadmsaadaot) [reply met-web/src/components/common/Modals/DayCalculator/index.tsx:80]: you can try this as well https://github.com/ryanmcdermott/clean-code-javascript#encapsulate-conditionals

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Let's try simplifying the code in DayCalculator. https://github.com/ryanmcdermott/clean-code-javascript is a good guide for some clean code guidelines

File: met-web/src/components/common/Modals/DayCalculator/index.tsx (hunk 1 above)
Comment (VineetBala-AOT) [reply met-web/src/components/common/Modals/DayCalculator/index.tsx]: updated as per review comments

File: met-web/src/components/common/Modals/DayCalculator/index.tsx (hunk 2 above)
Comment (VineetBala-AOT) [reply met-web/src/components/common/Modals/DayCalculator/index.tsx]: updated as per review comments

File: met-web/src/components/common/Modals/DayCalculator/index.tsx (hunk 3 above)
Comment (VineetBala-AOT) [reply met-web/src/components/common/Modals/DayCalculator/index.tsx]: updated as per review comments

**Pull request merged**